hydrolib/core/io/mdu/models.py

```diff
@@ -168,13 +166,37 @@ class Restart(INIBasedModel):
 class Boundary(INIBasedModel):
     _header: Literal["Boundary"] = "Boundary"
     quantity: str
-    nodeId: Optional[str]
+    nodeid: Optional[str]
+    locationfile: Optional[Path]
     forcingfile: Optional[ForcingModel] = None
     bndWidth1D: Optional[float]
 
     def is_intermediate_link(self) -> bool:
         return True
 
+    @root_validator
+    @classmethod
+    def check_nodeid_or_locationfile_present(cls, values: Dict):
+        """
+        Verifies that either nodeid or locationfile properties have been set.
+
+        Args:
+            values (Dict): Dictionary with values already validated.
+
+        Raises:
+            ValueError: When none of the values are present.
+
+        Returns:
+            Dict: Validated dictionary of values for Boundary.
+        """
+        node_id = values.get("nodeid", None)
+        forcingfile = values.get("forcingfile", None)
+        if str_is_empty_or_none(node_id) and not isinstance(forcingfile, ForcingModel):
+            raise ValueError(
+                "Either nodeid or locationfile fields should be specified."
```

Comment:
Please use correct casing in end user messages, so in this case: "Either nodeId or locationFile fields should be specified."

Reply:
in the same model these properties are defined in lower case.

Reply:
Modified properties names and messages (also in the tests).